pages/agent/v3/securing.md

```diff
@@ -104,6 +102,22 @@ do
 done
 ```
 
+You can see from the previous example that `$BUILDKITE_ENV_FILE` is the location of file that contains the environment variables that the control plane passes to a job. You may use this to block jobs from executing if certain environment variables are set.
+
+```bash
+#!/bin/bash
+
+set -euo pipefail
+
+if grep '^ENVIRONMENT_VARIABLE_TO_DENY=' "$BUILDKITE_ENV_FILE" > /dev/null
+then
+  echo "Rejecting job because it the environment variable ENVIRONMENT_VARIABLE_TO_DENY has been set"
+  exit 1
+fi
+```
+
+But remember to that some [environment variables may be essential](https://buildkite.com/docs/pipelines/environment-variables) to the execution of jobs, so deny-listing them in this manner is not advisable.
```

Comment:
This is mainly personal preference, so feel free to disagree. The one part that should change is using a relative link to the docs page.

```suggestion
But also remember that some [environment variables may be essential](/docs/pipelines/environment-variables) to the execution of jobs, so adding them to a blocklist in this manner is not advisable.
```